tcrypto/sys/types/build.rs

```diff
@@ -2,49 +2,47 @@
 //! Builds the FFI type bindings for the trusted crypto functions, (aes, rsa,
 //! etc.), of the Intel SGX SDK
 
-use bindgen::{callbacks::ParseCallbacks, Builder};
-
-#[derive(Debug)]
-struct Callbacks;
-
-impl ParseCallbacks for Callbacks {
-    fn item_name(&self, name: &str) -> Option<String> {
-        if name.starts_with("_sgx") {
-            Some(name[1..].to_owned())
-        } else {
-            None
-        }
-    }
-}
+const CRYPTO_TYPES: &[&str] = &[
+    "_sgx_ec256_dh_shared_t",
+    "_sgx_ec256_private_t",
+    "_sgx_ec256_public_t",
+    "_sgx_ec256_signature_t",
+    "_sgx_rsa3072_key_t",
+    "_sgx_rsa3072_public_key_t",
+    "sgx_aes_ctr_128bit_key_t",
+    "sgx_aes_gcm_128bit_key_t",
+    "sgx_aes_gcm_128bit_tag_t",
+    "sgx_aes_state_handle_t",
+    "sgx_cmac_128bit_key_t",
+    "sgx_cmac_128bit_tag_t",
+    "sgx_cmac_state_handle_t",
+    "sgx_ec256_shared_point_t",
+    "sgx_ecc_state_handle_t",
+    "sgx_hmac_state_handle_t",
+    "sgx_rsa3072_signature_t",
+    "sgx_rsa_key_type_t",
+    "sgx_rsa_result_t",
+    "sgx_sha1_hash_t",
+    "sgx_sha256_hash_t",
+    "sgx_sha384_hash_t",
+    "sgx_sha_state_handle_t",
+];
```

Comment:
What about types which are declared in `sgx_tcrypto.h` but not used by any of the exported functions? I'm OK with not exporting them but I thought I should bring them up.

* `sgx_hmac_256bit_key_t`
* `sgx_hmac_256bit_tag_t`
* `sgx_generic_ecresult_t`
* `rsa_params_t`

Reply:
I would say types in the header, even if not used should be there.  Longer term @jcape mentioned a desire to automate upgrades, and for this we would most likely be parsing the header.  
There are a few SGX apis that use void pointers, and then later cast to the type so relying on transitive functions probably isn't the best way.